mkosi/addon.py

```diff
@@ -0,0 +1,212 @@
+            "apt/sources.list.d/mkosi.sources",
+            "zypp/repos.d/mkosi.repo",
+            "pacman.conf",
+        ):
+            (Path(sandbox_tree) / "etc" / p).parent.mkdir(parents=True, exist_ok=True)
+            (Path(sandbox_tree) / "etc" / p).touch()
+
+        # Copy in the host's package manager configuration.
+        for p in (
+            "dnf",
+            "yum.repos.d/",
+            "apt",
+            "zypp",
+            "pacman.conf",
+            "pacman.d/",
+        ):
+            if not (Path("/etc") / p).exists():
+                continue
+
+            (Path(sandbox_tree) / "etc" / p).parent.mkdir(parents=True, exist_ok=True)
+            if (Path("/etc") / p).resolve().is_file():
+                shutil.copy2(Path("/etc") / p, Path(sandbox_tree) / "etc" / p)
+            else:
+                shutil.copytree(
+                    Path("/etc") / p,
+                    Path(sandbox_tree) / "etc" / p,
+                    ignore=shutil.ignore_patterns("gnupg"),
+                    dirs_exist_ok=True,
+                )
+
+        cmdline += ["--sandbox-tree", sandbox_tree]
+
+        # Generate crypttab with all the x-initrd.attach entries
+        if Path("/etc/crypttab").exists():
+            crypttab = [
+                line
+                for line in Path("/etc/crypttab").read_text().splitlines()
+                if (
+                    len(entry := line.split()) >= 4
+                    and not entry[0].startswith("#")
+                    and "x-initrd.attach" in entry[3]
+                )
+            ]
+            print(crypttab)
+            if crypttab:
+                with (Path(staging_dir) / "crypttab").open("w") as f:
+                    f.write("# Automatically generated by mkosi-addon\n")
+                    f.write("\n".join(crypttab))
+                cmdline += ["--extra-tree", f"{staging_dir}/crypttab:/etc/crypttab"]
+
+        if Path("/etc/kernel/cmdline").exists():
+            cmdline += ["--kernel-command-line", Path("/etc/kernel/cmdline").read_text()]
+
+        # Prefer dnf as dnf5 has not yet officially replaced it and there's a much bigger chance that there
+        # will be a populated dnf cache directory.
+        run(
+            cmdline,
+            stdin=sys.stdin,
+            stdout=sys.stdout,
+            env={"MKOSI_DNF": dnf.name} if (dnf := find_binary("dnf", "dnf5")) else {},
+        )
+
+        if args.output_dir:
+            with umask(~0o700) if os.getuid() == 0 else cast(umask, contextlib.nullcontext()):
+                Path(args.output_dir).mkdir(parents=True, exist_ok=True)
+        else:
+            args.output_dir = Path.cwd()
+
+        log_notice(f"Copying {staging_dir}/{args.output} to {args.output_dir}/{args.output}")
+        # mkosi symlinks the expected output image, so dereference it
+        copy_tree(
+            Path(f"{staging_dir}/{args.output}").resolve(),
+            Path(f"{args.output_dir}/{args.output}"),
+        )
```

Comment:
I think we can share this with `mkosi-initrd` as well

Reply:
seems a bit overkill for one function

Reply:
I mean with the output dir shenanigans above as well